### DISCUSSION TITLE
Unabled to clone flux2, context deadline exceeded - alert

### DISCUSSION THREAD

**Discussion by iangregsondev** [fluxcd/flux2#2171] [Q&A] [answered]
Hi,

I recently received this on my alert from fluxcd. Was this just a temporary thing ?

```
gitrepository/monitoring.flux-system
unable to clone 'https://github.com/fluxcd/flux2', error: Get "https://github.com/fluxcd/flux2/info/refs?service=git-upload-pack": context deadline exceeded
```

Anybody else see the problem or is it just me :-)

Also tried clicking on this https://github.com/fluxcd/flux2/info/refs?service=git-upload-pack

and I get a 400, not sure if that's right.

Thanks.

**Comment by stefanprodan** [comment, marked as answer]
The "context deadline exceeded" error means that source-controller wasn't able to clone the repo in 20s (default timeout). My guess is that your cluster egress is having issues, either the link is down or the network is overloaded. The git pull will be retried, at the specified interval. There is nothing we can do in Flux if GitHub goes down or if your network is flaky. You can increase the timeout, set a faster pull interval.

**Reply by iangregsondev** [reply]
Thanks.

**Comment by maxkokocom** [comment]
Hi, I am also getting the same error, but I don't think it's network issue. I have two clusters with exact same GitRepository and exact same Secret. One has ClusterAPI the other is created via ClusterAPI. Once bootstrapped it has flux definitions which successfully creates flux and tries to pull other applications from the same repo, but I am getting this error. Any ideas on how to debug it and figure out what's wrong? 

GitRepository:
```yaml
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  creationTimestamp: "2021-12-15T12:05:13Z"
  finalizers:
  - finalizers.fluxcd.io
  generation: 1
  name: flux-system
  namespace: flux-system
  resourceVersion: "32909"
  uid: 6b7d2657-f465-4e93-8dc3-904755be5f56
spec:
  gitImplementation: go-git
  interval: 1m0s
  ref:
    branch: master
  secretRef:
    name: gitops
  timeout: 20s
  url: https://gitlab.com/XXX/gitops.git
status:
  conditions:
  - lastTransitionTime: "2021-12-15T14:18:14Z"
    message: 'unable to clone ''https://gitlab.com/XXX/gitops.git'': Get "https://gitlab.com/XXX/gitops.git/info/refs?service=git-upload-pack":
      context deadline exceeded'
    reason: GitOperationFailed
    status: "False"
    type: Ready
  observedGeneration: 1
```

My Flux version: 
```yaml
flux: v0.23.0
helm-controller: v0.13.0
kustomize-controller: v0.18.0
notification-controller: v0.18.1
source-controller: v0.18.0
```

**Comment by mschielmann** [comment]
I have a similar issue (controller log):
```
{"level":"error","ts":"2022-08-12T11:18:08.619Z","logger":"controller.gitrepository","msg":"failed to checkout and determine revision: unable to clone 'https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git': Get \"https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git/info/refs?service=git-upload-pack\": dial tcp {{IP}}:443: i/o timeout","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","name":"flux-system","namespace":"flux-system","reconcileID":"2d466f5f-bbd9-4a1d-9d53-0858e096c667","error":"failed to checkout and determine revision: unable to clone 'https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git': Get \"https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git/info/refs?service=git-upload-pack\": dial tcp {{IP}}:443: i/o timeout"}
```
when bootstraping flux with the below command:
```
flux bootstrap gitlab --hostname={{HOSTNAME}} --ssh-hostname={{HOSTNAME}} --owner={{GROUP_NAME}} --repository={{REPO_NAME}} --branch=master --path=clusters/staging --secret-name={{SECRET_NAME}} --namespace=flux-system
```
The secret contains a private key info for ssh.
Basically this https calls requires basic auth. I'm not sure why this is needed since I can see that flux is making pushes to that repo anyway. 
I've tried exporting GITLAB_TOKEN (PAT), GITLAB_USER/GITLAB_PASSWORD. By the way, I'n not sure why HTTPS is used / shown everywhere, I'd have preferred it used SSH.

Complete bootstrap output:
► connecting to https://{{HOSTNAME}}
► cloning branch "master" from Git repository "https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git"
✔ cloned repository
► generating component manifests
✔ generated component manifests
✔ committed sync manifests to "master" ("6e4b37daf7b1426da2b111d31bd4354cd98e542c")
► pushing component manifests to "https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git"
► installing components in "flux-system" namespace
✔ installed components
✔ reconciled components
► determining if source secret "flux-system/{{SECRET_NAME}}" exists
✔ source secret up to date
► generating sync manifests
✔ generated sync manifests
✔ committed sync manifests to "master" ("e0d00fc91727f30437a54cd26a6e4dfb821d61d0")
► pushing sync manifests to "https://{{HOSTNAME}}/{{GROUP_NAME}}/{{REPO_NAME}}.git"
► applying sync manifests
✔ reconciled sync configuration
◎ waiting for Kustomization "flux-system/flux-system" to be reconciled
✗ client rate limiter Wait returned an error: context deadline exceeded
► confirming components are healthy
✔ helm-controller: deployment ready
✔ kustomize-controller: deployment ready
✔ notification-controller: deployment ready
✔ source-controller: deployment ready
✔ all components are healthy
✗ bootstrap failed with 1 health check failure(s)

$ flux --version
flux version 0.31.5

Any help would be greatly appreciated - please let me know if I should create a separate issue or add more info to this one.

**Reply by mr-pascal** [reply]
@mschielmann did you somehow manage to get around this?
I have also the same issue, that I run into the "client rate limiter" issue...

**Reply by mschielmann** [reply]
No, I gave up on this project. For me it was just a trial and I didn't find it helpful / useful, I gave up on it.

**Reply by mr-pascal** [reply]
not 100% sure if this solved my issue (but at least it worked afterward).
It am running/testing on MicroK8s on my local machine.
And after adding coreDNS ( `microk8s enable dns`) it seems to work. I think the initial issue was that networking wasn't working as expected and adding DNS to my cluster solved the connection issues.
